src/main.js

```diff
@@ -13,22 +13,71 @@
  *
  */
 
-// Does ECN Viewer need port 80 ?
-const requireElevated = process.argv[2] === 'start' && (!process.env.VIEWER_PORT || process.env.VIEWER_PORT === '80')
-
+const Cli = require('./cli')
+const daemon = require('./daemon')
+const config = require('./config')
 const isElevated = require('is-elevated')
+const request = require('request-promise')
+
+const isHTTPS = () => {
+  const sslKey = config.get('Server:SslKey')
+  const devMode = config.get('Server:DevMode')
+  const sslCert = config.get('Server:SslCert')
+  const intermedKey = config.get('Server:IntermediateCert')
+  return !devMode && sslKey && sslCert && intermedKey
+}
+
+const getJSONFromURL = async (uri) => request({
+  uri,
+  json: true
+})
+
+const isDaemonElevated = async () => {
+  // If it is running and you can see it, you have enough permission to move forward
+  if (daemon.status() !== 0) {
+    return false
+  }
+  const apiPort = config.get('Server:Port')
+  const protocol = isHTTPS() ? 'https' : 'http'
+  return getJSONFromURL(`${protocol}://localhost:${apiPort}/api/v3/status`)
+    .then(result => {
+      // The server is running but you couldn't see it, requires administrative privileges
+      if (result.status === 'online') {
+        return true
+      }
+      // The server is not running
+      return false
+    })
+    .catch((err) => {
+      // If error is connection refused, server is not running
+      if (err.error && err.error.code === 'ECONNREFUSED') { return false }
+      // If error, defaults to requiring administrative privileges
+      return true
+    })
+}
+const elevatedCommands = ['start', 'stop', 'controller status']
+const requiresElevated = async (command, runningAsRoot) => {
+  // Does ECN Viewer need port 80 ?
+  const viewerPort = +(process.env.VIEWER_PORT || config.get('Viewer:Port'))
+  if (process.argv[2] === 'start' && (viewerPort === 80)) {
```

Comment:
What if viewer port is 81?
We need service to be started with `sudo` if either viewer port or controller port is running on a port < 1024